crates/model/src/auction.rs

```diff
@@ -0,0 +1,70 @@
+//! Module defining a batch auction.
+
+use crate::{order::Order, u256_decimal::DecimalU256};
+use primitive_types::{H160, U256};
+use serde::{Deserialize, Serialize};
+use serde_with::serde_as;
+use std::collections::BTreeMap;
+
+/// A batch auction.
+#[serde_as]
+#[derive(Clone, Debug, Deserialize, Serialize, PartialEq)]
+pub struct Auction {
+    /// The block that this auction is valid for.
+    /// The block number for the auction. Orders and prices are guaranteed to be
+    /// valid on this block.
+    pub block: u64,
+
+    /// The solvable orders included in the auction.
+    pub orders: Vec<Order>,
+
+    /// The reference prices for all traded tokens in the auction.
+    #[serde_as(as = "BTreeMap<_, DecimalU256>")]
+    pub prices: BTreeMap<H160, U256>,
```

Comment:
We used `HashMap` for clearing prices in the `Settlement`, any particular reason to use `BTreeMap` here?

Reply:
I chose `BTreeMap` just to make sure that the JSON output is always ordered. So multiple calls to `v1/auction` return the same JSON blob (and don't depend on the randomness of `HashMap` and its use of `Hash`).